### PR TITLE
Az OIDC

### PR DESCRIPTION
**Pull request by TylerMizuyabu**
Adds module to setup azure resources necessary for an oidc connection for terraform

Adds support to configure github vars to use az oidc connection

### CODE REVIEW

File: modules/github-azure-oidc/oidc.tf [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+data "azurerm_key_vault" "key_vault" {
+  count =  var.kv_name != "" ? 1 : 0
+  name                = var.kv_name
+  resource_group_name = var.kv_resource_group != "" ? var.kv_resource_group : local.github_foundations_rg.name
+}
+
+/**
+* User assigned identities and roles for github state bucket and federated identity setup
+*/
+
```
Comment (bzarboni1) [inline modules/github-azure-oidc/oidc.tf:49]:
```suggestion
```

File: modules/github-azure-oidc/oidc.tf [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        role_definition_name = "Key Vault Reader"
+      }
+    }: {}
+  )
+}
+
+data "azurerm_client_config" "current" {}
+
+data "azurerm_key_vault" "key_vault" {
+  count =  var.kv_name != "" ? 1 : 0
```
Comment (bzarboni1) [inline modules/github-azure-oidc/oidc.tf:41]:
```suggestion
  count                =  var.kv_name != "" ? 1 : 0
```

File: modules/github-azure-oidc/storage.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  count              = var.tf_state_container_default_encryption_scope.name != "" ? 1 : 0
+  name               = var.tf_state_container_default_encryption_scope.name
+  storage_account_id = azurerm_storage_account.github_foundations_sa.id
+  source             = var.tf_state_container_default_encryption_scope.source
+  key_vault_key_id   = var.tf_state_container_default_encryption_scope.key_vault_key_id
+}
+
+resource "azurerm_storage_container" "github_foundations_tf_state_container" {
+  count = local.default_encryption_scope == null ? 1 : 0 
+  name                              = var.tf_state_container
```
Comment (bzarboni1) [inline modules/github-azure-oidc/storage.tf:24]:
```suggestion
  count                             = local.default_encryption_scope == null ? 1 : 0 
  name                              = var.tf_state_container
```

File: modules/github-azure-oidc/storage.tf [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  storage_account_name              = azurerm_storage_account.github_foundations_sa.name
+  container_access_type             = var.tf_state_container_anonymous_access_level
+}
+
+resource "azurerm_storage_container" "github_foundations_tf_state_encrypted_container" {
+  count = local.default_encryption_scope != null ? 1 : 0
+  name                              = var.tf_state_container
```
Comment (bzarboni1) [inline modules/github-azure-oidc/storage.tf:31]:
```suggestion
  count                             = local.default_encryption_scope != null ? 1 : 0
  name                              = var.tf_state_container
```

File: modules/github-azure-oidc/variables.tf [hunk 5]
```diff
@@ -0,0 +1,92 @@
+#Resource Group Variables
+variable "rg_create" {
+  description = "Create resource group. When set to false, uses id to reference an existing resource group."
```
Comment (bzarboni1) [inline modules/github-azure-oidc/variables.tf:3]:
```suggestion
  description = "Create the resource group. When set to false, it uses (the/an?) id to reference an existing resource group."
```

File: modules/github-azure-oidc/variables.tf [hunk 6]
```diff
@@ -0,0 +1,92 @@
+#Resource Group Variables
```
Comment (bzarboni1) [inline modules/github-azure-oidc/variables.tf:1]:
```suggestion
# Resource Group Variables
```

File: modules/github-azure-oidc/variables.tf [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+variable "sa_tier" {
+  type        = string
+  description = "The tier of the storage account for github foundations. Valid options are Standard and Premium."
+}
+
+variable "sa_replication_type" {
+  type        = string
+  description = "The replication type of the storage account for github foundations. Valid options are LRS, GRS, RAGRS, ZRS, GZRS, and RA_GZRS."
+}
```
Comment (bzarboni1) [inline modules/github-azure-oidc/variables.tf:32]: Should we select a sensible default value here?

File: modules/github-foundations/azure-oidc-variables.tf [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+  ]
+}
+
+resource "github_actions_organization_variable" "storage_account_name" {
+  count = var.oidc_configuration.azure != null ? 1 : 0
+
+  variable_name     = coalesce(var.oidc_configuration.azure.storage_account_name_variable_name, "AZURE_STORAGE_ACCOUNT_NAME")
+  value = var.oidc_configuration.azure.storage_account_name
+  visibility      = "selected"
+  selected_repository_ids = [
```
Comment (bzarboni1) [inline modules/github-foundations/azure-oidc-variables.tf:59]:
```suggestion
  variable_name     = coalesce(var.oidc_configuration.azure.storage_account_name_variable_name, "AZURE_STORAGE_ACCOUNT_NAME")
  value             = var.oidc_configuration.azure.storage_account_name
  visibility        = "selected"
  selected_repository_ids = [
```

File: modules/github-foundations/azure-oidc-variables.tf [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+  ]
+}
+
+resource "github_actions_organization_variable" "container_name" {
+  count = var.oidc_configuration.azure != null ? 1 : 0
+
+  variable_name     = coalesce(var.oidc_configuration.azure.container_name_variable_name, "AZURE_CONTAINER_NAME")
+  value = var.oidc_configuration.azure.container_name
+  visibility      = "selected"
+  selected_repository_ids = [
```
Comment (bzarboni1) [inline modules/github-foundations/azure-oidc-variables.tf:71]:
```suggestion
  variable_name     = coalesce(var.oidc_configuration.azure.container_name_variable_name, "AZURE_CONTAINER_NAME")
  value             = var.oidc_configuration.azure.container_name
  visibility        = "selected"
  selected_repository_ids = [
```

**Review by bzarboni1 — APPROVED** [review]
Looks really good! Super impressed how quickly you pulled this together!

Just a few syntax suggestions, and one question / comment. Feel free to disregard any of them.

**Review by bzarboni1 — APPROVED** [review]
Looks really good! Super impressed how quickly you pulled this together!

Just a few syntax suggestions, and one question / comment. Feel free to disregard any of them.

File: modules/github-azure-oidc/variables.tf (hunk 7 above)
Comment (TylerMizuyabu) [reply modules/github-azure-oidc/variables.tf]: Set to Standard and GRS. The defaults you get in the az portal

**Review by bzarboni1 — APPROVED** [review]

**Pull request merged**